File: benchmark/bench_arith.py
n = 2000000
mod = 100000
a = 1
b = 2
c = 3
total = 0
i = 0

while i < n:
    a = (a + i + 7) % mod
    b = (b + a * 3 + 11) % mod
    c = (c + a + b) % mod
    total = (total + a + b + c) % mod
    i += 1

print(total + a + b + c)

File: benchmark/bench_arith.lua
local function run()
    local n = 5000000
    local i = 0
    local a = 1
    local b = 2
    local c = 3
    local total = 0

    while i < n do
        a = (a + i + 7) % 1000
        b = (b + a * 3 + 11) % 1000
        c = (c + a + b) % 1000

        if c > 500 then
            total = total + c - a
        else
            total = total + b - c
        end

        i = i + 1
    end

    return total + a + b + c
end

print(run())
